evaluator/src/main/kotlin/PackageRule.kt

```diff
@@ -70,6 +70,31 @@ open class PackageRule(
         licenseRules.forEach { it.evaluate() }
     }
 
+    /**
+     * A [RuleMatcher] that checks whether any severity of the [package][pkg] using the specified [scoringSystem] is
+     * higher than [maxValue]. Tested using the [severityComparator]. Returns false if the severity is equal to
+     * [maxValue].
+     */
+    fun containsSeverityOver(maxValue: String, scoringSystem: String, severityComparator: Comparator<String>) =
```

Comment:
@MarcelBochtler  I would rename "containsSeverityOver" to "hasVulnerabilityWithSeverityOver" to align the function name with existing package rulematchers

Reply:
Renamed it again to `hasVulnerability` so it can also be a little bit more generalised.
See also https://github.com/oss-review-toolkit/ort/pull/3961#discussion_r623162217